### DISCUSSION TITLE
Can I store view in a variable that I would wrap?

### DISCUSSION THREAD

**Discussion by Umenokin** [leptos-rs/leptos#1644] [Q&A] [answered]
I read [https://leptos-rs.github.io/leptos/view/06_control_flow.html](Control Flow) chapter from the book but could not implement something that compiles. Is there a way to do something like this?

```
#[component]
pub fn Logo(
    #[prop(optional)]
    href: Option<&'static str>,
) -> impl IntoView {
   let content = view! { SOME RELATIVELY BIG HTML LIKE SVG }

    view! {
        {move || {
            match href {
                Some(href) =>  view! {
                    <A href={href}>{content}</A>
                }.into_view(),
                None => view! {
                    {content}
                }.into_view()
            }
        }}
    }
}

```

Basically, I will render content 100% but might wrap it or not. And content potentially might be bigger than a simple string thus, want to avoid manually inlining it (What I am am currently doing)

Thanks!

**Comment by gbj** [comment, marked as answer]
It would be helpful to see an example of your actual code that doesn't compile, and to know what the compile errors are. What you're trying to do is perfectly achievable; here would be one example that does compile:
```rust
#[component]
pub fn Logo(#[prop(optional)] href: Option<&'static str>) -> impl IntoView {
    let content = view! { <p>"ok"</p> };

    move || match href {
        Some(href) => {
            let content = content.clone();
            view! {
                <A href={href}>{content}</A>
            };
        }
        .into_view(),
        None => content.clone().into_view(),
    }
}
```
Note, by the way, that several of your `view!` invocations are unnecessary: the `move || match` closure and `view! { {content} }` do not need to be wrapped in a macro, they can just be their plain selves.

**Reply by Umenokin** [reply]
Thank you. I example is actually working. I think just just used too many `view!` and also did not use clone
